packages/fungible-token/bridge-fungible-token/src/main.sw

```diff
@@ -201,28 +203,45 @@ impl Bridge for Contract {
 impl SRC20 for Contract {
     #[storage(read)]
     fn total_assets() -> u64 {
-        0u64
+        storage.total_assets.try_read().unwrap_or(0)
     }
 
     #[storage(read)]
-    fn total_supply(_asset: AssetId) -> Option<u64> {
-        None
+    fn total_supply(asset: AssetId) -> Option<u64> {
+        if storage.asset_to_sub_id.get(asset).try_read().is_none() {
+            return None;
+        }
+
+        // Reasoning: an extreme supply token such as SHIBA could overflow the
+        // total supply, which is defined by SRC20 as u64.
+        // .as_u64() will revert on overflow, in that case, it will return u64::max() 
+        Some(storage.tokens_minted.get(asset).read().as_u64().unwrap_or(u64::max()))
```

Comment:
So this will technically be an incorrect value under an overflow. Unsure about this

Reply:
Fixed: it now registers a refund when the total_supply cap is reached